### PR TITLE
Trim long URLs, paths and other list items uniformly

### PR DESCRIPTION
**Pull request by aerosol**
### Changes

This PR extends the `trimURL` to fall back to regular strings in case page paths are provided. It's now applied on all the lists and modals.

image

image

image

image

image

image

image

### Tests
- [ ] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [ ] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [ ] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [ ] This PR does not change the UI

### CODE REVIEW

File: assets/js/dashboard/stats/modals/entry-pages.js [hunk 1]
```diff
@@ -78,15 +79,15 @@ class EntryPagesModal extends React.Component {
...
+        <td className="p-2 truncate">
           <Link
             to={{
               pathname: `/${encodeURIComponent(this.props.site.domain)}`,
               search: query.toString()
             }}
             className="hover:underline"
           >
-            {page.name}
+            {trimURL(page.name, 40)}
```
Comment (aerosol) [inline assets/js/dashboard/stats/modals/entry-pages.js:90]: the actual change

File: assets/js/dashboard/stats/modals/entry-pages.js [hunk 2]
```diff
@@ -78,15 +79,15 @@ class EntryPagesModal extends React.Component {
 
     return (
       <tr className="text-sm dark:text-gray-200" key={page.name}>
-        <td className="p-2">
+        <td className="p-2 truncate">
```
Comment (aerosol) [inline assets/js/dashboard/stats/modals/entry-pages.js:82]: the actual change

File: assets/js/dashboard/stats/modals/exit-pages.js [hunk 3]
```diff
@@ -68,8 +68,8 @@ class ExitPagesModal extends React.Component {
 
     return (
       <tr className="text-sm dark:text-gray-200" key={page.name}>
-        <td className="p-2">
-          <Link to={{pathname: `/${encodeURIComponent(this.props.site.domain)}`, search: query.toString()}} className="hover:underline">{page.name}</Link>
+        <td className="p-2 truncate">
+          <Link to={{ pathname: `/${encodeURIComponent(this.props.site.domain)}`, search: query.toString() }} className="hover:underline">{trimURL(page.name, 40)}</Link>
```
Comment (aerosol) [inline assets/js/dashboard/stats/modals/exit-pages.js:72]: the actual change

File: assets/js/dashboard/stats/modals/pages.js [hunk 4]
```diff
@@ -63,14 +64,14 @@ class PagesModal extends React.Component {
     return (
       <tr className="text-sm dark:text-gray-200" key={page.name}>
         <td className="p-2">
-          <Link to={{pathname: `/${encodeURIComponent(this.props.site.domain)}`, search: query.toString()}} className="hover:underline block truncate">{page.name}</Link>
+          <Link to={{ pathname: `/${encodeURIComponent(this.props.site.domain)}`, search: query.toString() }} className="hover:underline block truncate">{trimURL(page.name, 50)}</Link>
```
Comment (aerosol) [inline assets/js/dashboard/stats/modals/pages.js:67]: the actual change

File: assets/js/dashboard/stats/reports/list.js [hunk 5]
```diff
@@ -245,7 +245,7 @@ export default function ListReport(props) {
               {maybeRenderIconFor(listItem)}
 
               <span className="w-full md:truncate">
-                {listItem.name}
+                {trimURL(listItem.name, 60)}
```
Comment (aerosol) [inline assets/js/dashboard/stats/reports/list.js:248]: the actual change

File: assets/js/dashboard/util/url.js [hunk 6]
```diff
@@ -64,6 +66,13 @@ export function trimURL(url, maxLength) {
...
 
     return trimmedURL;
+  } else {
+    const leftSideLength = Math.floor(maxLength / 2);
+    const rightSideLength = maxLength - leftSideLength;
+
+    const leftSide = url.slice(0, leftSideLength);
+    const rightSide = url.slice(-rightSideLength);
+
+    return leftSide + ellipsis + rightSide;
```
Comment (aerosol) [inline assets/js/dashboard/util/url.js:76]: the actual changes

**Review by vinibrsl — APPROVED** [review]

**Pull request merged**